listenbrainz_spark/stats/user/utils.py

```diff
@@ -25,30 +23,23 @@ def get_artists(table):
                 }
     """
     t0 = time.time()
-    result = run_query("""
-            SELECT user_name
-                 , artist_name
-                 , artist_msid
-                 , artist_mbids
-                 , count(artist_name) as cnt
-              FROM %s
-          GROUP BY user_name, artist_name, artist_msid, artist_mbids
-          ORDER BY cnt DESC
-        """ % (table))
-    rows = result.collect()
+    df = df.select("user_name","artist_name","artist_msid","artist_mbids")
```

Comment:
It will be good to not write a single query which includes all the sub queries.
Let us consider the below query:

`df = df.select("user_name", "artist_name")`
`df = df.groupBy("user_name, "artist_name"")`

I'd prefer this:
`artist_df = df.select("user_name", "artist_name") \`
`                    .groupBy("user_name", "artist_name)`

https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz_spark/recommendations/candidate_sets.py#L52